01_fib/starfallz/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, strings.Trim(strings.Join(strings.Split(fmt.Sprint(fib(7)), " "), "\n"), "[]"))
+}
+
+func fib(n int) []int {
+	if n == 0 {
+		return []int{0}
+	}
+
+	isNegativeFib := false
+
+	if n < 0 {
+		isNegativeFib = true
+		n *= -1
+	}
+
+	result := []int{1}
```

Comment:
`result := make([]int, n)` - you already know how big you need your slice to be.
slightly shorter:
```go
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	result := make([]int, n*sign)
	for i := 0; i < n*sign; i++ {
		n1, n2 = n2, n1+sign*n2
		result[i] = n1
	}
	return result
}

if you print inside the function as required you should get a shortened:

```go
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		fmt.Fprintln(out, n1)
	}
```

Reply:
Haha you are literally giving me the solution Julia 😂😂😂😂 I agree it's much shorter but should I be copy pasting your code!? Lol feels like it's a completely new implementation haha

Reply:
I know - can't help myself ;)

**TL;DR: You don't need to update anything (but you certainly can if you want to;) ).**

Some of us running this course have spent a bit of time going back and forth between us on what we think the canonical solution should be, and I always find it interesting to see how different the same short piece of functionality can look.  

Personally, I love it when others look at my code and help me make it shorter, nicer and/or more readable. Everything that makes it better makes _me_  happy: removing little stumbling blocks like typos, bad variable names, going against conventions and idioms or poor naming. Even more important I really appreciate it when medium size weirdnesses on a block/logic level are clarified or even bigger poor design choices are brought to light. 

To me it's not just about getting the job/ticket done but to have fun. To achieve this I see my job as a craft that wants to express itself as beautifully as it can and it almost always becomes better with many eyes looking over it. That's why I posted it :)

I guess that was the long answer.